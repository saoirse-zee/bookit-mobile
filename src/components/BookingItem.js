import React from 'react'
import { StyleSheet, Text, View, TouchableHighlight } from 'react-native'
import { formatDate } from '../utils'
import colors from '../../constants/Colors'


const BookingItem = ({
  booking,
  location,
  bookableName,
  onPressItem,
  selected,
}) => {
  const start = formatDate(booking.start, location.timeZone)
  const end = formatDate(booking.end, location.timeZone)

  return (
    <TouchableHighlight
      onPress={() => onPressItem(booking.id)}
      underlayColor="#ddd"
    >
      <View style={[
        styles.booking,
        booking.id === selected ? styles.selected : styles.notSelected]}
      >
        <Text style={styles.bookingName}>{bookableName} in {location.name}</Text>
        <Text style={styles.date}>Start: {start}</Text>
        <Text style={styles.date}>End: {end}</Text>
      </View>
    </TouchableHighlight>
  )
}

export default BookingItem

const styles = StyleSheet.create({
  booking: {
    marginBottom: 15,
  },
  bookingName: {
    fontSize: 18,
  },
  date: {
    fontSize: 13,
  },
  notSelected: {
    borderLeftColor: '#eee',
  },
  selected: {
    borderLeftColor: colors.tintColor,
  },
})
