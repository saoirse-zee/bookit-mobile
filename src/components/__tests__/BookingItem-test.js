import 'react-native'
import React from 'react'
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer'
import BookingItem from '../BookingItem'

const stubPressHandler = () => {}
let booking

describe('BookingItem', () => {
  beforeEach(() => {
    booking = {
      id: 'BOOKING-ID-44',
      formattedStart: 'January 1, 2017 2:30 PM PST',
      formattedEnd: 'January 1, 2017 3:15 PM PST',
      ownerName: 'Johnny Popsqueak',
      bookableName: 'Leica FaceZoomer XLT-Max',
      locationName: 'NYC',
    }
  })

  it('displays the given booking', () => {
    const component = (
      <BookingItem
        booking={booking}
        onPressItem={stubPressHandler}
        selected="BOOKING-ID-37"
      />
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('highlights a selected booking by coloring the left border blue', () => {
    const component = (
      <BookingItem
        booking={booking}
        onPressItem={stubPressHandler}
        selected="BOOKING-ID-44"
      />
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('hides the owner name if the owner is not given', () => {
    booking.ownerName = undefined
    const component = (
      <BookingItem
        booking={booking}
        onPressItem={stubPressHandler}
        selected="BOOKING-ID-44"
      />
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
