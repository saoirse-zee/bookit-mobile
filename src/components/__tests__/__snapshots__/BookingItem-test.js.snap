// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingItem displays the given booking 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 15,
        },
        Object {
          "borderLeftColor": "#eee",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Leica FaceZoomer XLT-Max
       in 
      NYC
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 13,
        }
      }
    >
      Start: 
      January 1, 2017 2:30 PM PST
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 13,
        }
      }
    >
      End: 
      January 1, 2017 3:15 PM PST
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 13,
        }
      }
    >
      Booked by: 
      Johnny Popsqueak
    </Text>
  </View>
</View>
`;

exports[`BookingItem hides the owner name if the owner is not given 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 15,
        },
        Object {
          "borderLeftColor": "#4F97FF",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Leica FaceZoomer XLT-Max
       in 
      NYC
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 13,
        }
      }
    >
      Start: 
      January 1, 2017 2:30 PM PST
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 13,
        }
      }
    >
      End: 
      January 1, 2017 3:15 PM PST
    </Text>
  </View>
</View>
`;

exports[`BookingItem highlights a selected booking by coloring the left border blue 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 15,
        },
        Object {
          "borderLeftColor": "#4F97FF",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Leica FaceZoomer XLT-Max
       in 
      NYC
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 13,
        }
      }
    >
      Start: 
      January 1, 2017 2:30 PM PST
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 13,
        }
      }
    >
      End: 
      January 1, 2017 3:15 PM PST
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 13,
        }
      }
    >
      Booked by: 
      Johnny Popsqueak
    </Text>
  </View>
</View>
`;
